optimum_benchmark/benchmarks/inference/benchmark.py

```diff
@@ -77,16 +79,24 @@ def run_forward_tracking(self, backend: "Backend") -> None:
         if self.config.memory:
             LOGGER.info("\t+ Tracking forward pass peak memory")
             memory_tracker = memory_tracker_class_for_backend[backend.config.name](backend)
-            with memory_tracker.track(interval=self.config.duration // 100):
+            with memory_tracker.track(interval=self.forward_latency / 10):
                 _ = backend.forward(forward_input)
 
             self.forward_peak_memory = memory_tracker.get_peak_memory()
             LOGGER.info(f"\t+ Forward pass peak memory: {self.forward_peak_memory} (MB)")
 
+        if self.config.emissions:
+            LOGGER.info("\t+ Tracking forward pass emissions")
+            emissions_tracker = EmissionsTracker()
+            with emissions_tracker.track(interval=self.forward_latency / 10, file_prefix="forward"):
```

Comment:
So we should have more or less 10 measures during the forward pass right?

Reply:
Yes just as rule of thumb I use the same value for memory measures. It should be noted that the refresh rate of power measures coming from `pynvml` is much slower.

Reply:
it seems the first option won't work because CodeCarbon uses a thread which interferes with inference (due to the GIL).

Reply:
I would definitely not measure latency and energy in the same forward pass. So the 2nd solution seems better to me.